lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +294,130 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
+
+    connection.execute("SELECT #{_quoted_partman_schema}.create_parent(#{create_parent_sql})")
+
+    update_config_options = {
+      infinite_time_partitions: infinite_time_partitions,
+      inherit_privileges: inherit_privileges,
+      retention: retention,
+      retention_keep_table: retention_keep_table,
+    }.compact
+
+    unsafe_partman_update_config(fully_qualified_table_name, **update_config_options)
+  end
+
+  def safe_partman_update_config(table, **options)
+    if options[:retention].present? || options[:retention_keep_table] == false
+      raise PgHaMigrations::UnsafeMigrationError.new(":retention and/or :retention_keep_table => false can potentially result in data loss if misconfigured. Please use unsafe_partman_update_config if you want to set these options")
+    end
+
+    unsafe_partman_update_config(table, **options)
+  end
+
+  def unsafe_partman_update_config(table, **options)
+    invalid_options = options.keys - PARTMAN_UPDATE_CONFIG_OPTIONS
+
+    raise ArgumentError, "Unrecognized argument(s): #{invalid_options}" unless invalid_options.empty?
+
+    PgHaMigrations::PartmanConfig.schema = _quoted_partman_schema
+
+    config = PgHaMigrations::PartmanConfig.find(_fully_qualified_partman_name(table))
+
+    config.assign_attributes(**options)
+
+    inherit_privileges_changed = config.inherit_privileges_changed?
+
+    config.save!
+
+    safe_partman_reapply_privileges(table) if inherit_privileges_changed
+  end
+
+  def safe_partman_reapply_privileges(table)
+    connection.execute("SELECT #{_quoted_partman_schema}.reapply_privileges('#{_fully_qualified_partman_name(table)}')")
+  end
+
+  def _quoted_partman_schema
+    schema = connection.select_value(<<~SQL)
+      SELECT nspname
+      FROM pg_namespace JOIN pg_extension
+        ON pg_namespace.oid = pg_extension.extnamespace
+      WHERE pg_extension.extname = 'pg_partman'
+    SQL
+
+    raise PgHaMigrations::InvalidMigrationError, "The pg_partman extension is not installed" unless schema.present?
+
+    connection.quote_schema_name(schema)
+  end
+
+  def _fully_qualified_partman_name(table)
+    table.to_s.split(".").each { |part| _validate_partman_identifier(part) }
+
+    return table if table.to_s.include?(".")
+
+    schemas = connection.select_values(<<~SQL)
+      SELECT schemaname
+      FROM pg_tables
+      WHERE tablename = '#{table}' AND schemaname = ANY (current_schemas(false))
+    SQL
+
+    raise PgHaMigrations::InvalidMigrationError, "Could not find #{table} in search path" unless schemas.present?
+    raise PgHaMigrations::InvalidMigrationError, "Found #{table} in multiple schemas: #{schemas}" if schemas.size > 1
+
+    _validate_partman_identifier(schemas.first)
+
+    "#{schemas.first}.#{table}"
+  end
+
+  def _validate_partman_identifier(identifier)
+    if identifier.to_s !~ /^[a-z_][a-z_\d]*$/
+      raise PgHaMigrations::InvalidMigrationError, "Partman requires schema / table names to be lowercase with underscores"
```

Comment:
You previously made a comment about escaping the values we pass to `create_parent` and that made me realize we weren't sanitizing table names in any way. Since we're calling partman functions and passing the table name as a string and aren't just referencing it as an identifier in a normal query, we can't use things like `quote_table_name`. Now, I also tried using things like `quote` and `escape_string`, but that generated all sorts of errors in the internal partman functions. What I've gathered is partman has very limited support for tables with non-standard names, so I think it seems valuable to enforce that in our code to avoid strange errors from partman

Reply:
That all makes sense, except that probably it's OK with uppercase? Lowercase is, of course, the Rails convention, so I don't feel strongly about supporting that, but still curious.

Reply:
In the absence of quoting (because partman doesn't like it), the use of uppercase letters would cause some confusion as I believe they would just get converted to lowercase. That would also be inconsistent with other Rails methods so I'd prefer to completely disallow uppercase